### ISSUE TITLE
Order of raising resize event relative to updating bounds

### ISSUE THREAD

**Issue by pearcetm** [openseadragon/openseadragon#2281]
I noticed something funny and somewhat counterintuitive related to a discussion in another thread: https://github.com/openseadragon/openseadragon/discussions/2279

If you want to take a navigation action in response to a `resize` event, you currently cannot do this directly in a function bound to the `resize` event - it must be deferred via `window.setTimeout()`, or by a similar mechanism. Why?
```
        if( this.viewer ){
            /**
             * Raised when the viewer is resized (see {@link OpenSeadragon.Viewport#resize}).
             *
             * @event resize
             * @memberof OpenSeadragon.Viewer
             * @type {object}
             * @property {OpenSeadragon.Viewer} eventSource - A reference to the Viewer which raised this event.
             * @property {OpenSeadragon.Point} newContainerSize
             * @property {Boolean} maintain
             * @property {?Object} userData - Arbitrary subscriber-defined object.
             */
            this.viewer.raiseEvent( 'resize', {
                newContainerSize: newContainerSize,
                maintain: maintain
            });
        }

        return this.fitBounds( newBounds, true ); //overrides any navigation triggered in a resize handler
```

The `resize` event is fired, and only afterwards does the viewer go to the new computed bounds. Any navigation actions taken in the `resize` handler/callback are immediately overridden!

I feel like it would make the most sense to swap the order of these operations. However, I can imagine that might break existing apps if they depend on knowing the `resize` is in progress in order to listen for `viewport-change` or similar, during the subsequent `fitBounds` call. An alternative could be firing a new event `after-resize` which would be raised be added after `fitBounds` is called, providing a hook for customizing navigation on resizing the viewer.

**Comment by iangilman** [comment]
Excellent point! I agree that the safest approach would be to add a new event, rather than changing the current one. On the one hand, I'm not entirely sure what scenarios would require knowing about the resize before it's processed, so maybe it's not really necessary to have a new event. On the other hand, it's not really a big deal to add an additional event, so we might as well.

**Comment by iangilman** [comment]
I've been pondering this, and I think I'm leaning toward adding a new `after-resize` event. Changing when the `resize` event happens is technically a breaking change, so if we do it, we'll need to bump the version for the next release up to 5.0.0, and it seems like such a minor thing to want to do that for. Also, since this issue hasn't come up before, I don't think it's a really hot topic for most people.

I like the terminology "after" because it has an obvious pair in "before" if we ever needed to add such for one of our events. That said, we already have "pre-full-screen" whose pair would be "post". So maybe we should go with `post-resize`? Before/after feels more friendly to me, and we only have that one, so I don't know that we need to set pre/post into stone. 

There I go again, thinking out loud! I hope you don't mind being part of this decision making process :D

**Comment by pearcetm** [comment]
I'm happy with `after-resize`. If it's not already done in the next week or two I should have a chance to put a PR together.

**Comment by iangilman** [comment]
Great! :)

**Comment by iangilman** [comment]
Fixed by #2317
